lib/ui/painting.dart

```diff
@@ -154,6 +154,27 @@ class Color {
     return new Color.fromARGB(alpha, red, green, b);
   }
 
+  /// Apply to canonical sRGB transfer function to convert to linear values.
+  static double _srgbToLinear(int color) {
+    assert(0 <= color && color <= 255);
+
+    double x = color * (1.0 / 255.0);
+    if (x <= 0.04045)
+      return x * (1.0 / 12.92);
+
+    return math.pow((x + 0.055) * (1.0 / 1.055), 2.4);
+  }
+
+  /// Apply the canonical inverse sRGB transfer function to convert from linear values.
+  static int _srgbFromLinear(double x) {
+    assert(0.0 <= x && x <= 1.0);
```

Comment:
BTW this assumption is not valid. In particular, the `t` passed to `lerp` could be outside the range 0.0-1.0. (I discovered this when running this code in a quick benchmark.)